docs(common): Styling and Themes revamp (#2205)
telerik/blazor-docs: knowledge-base/common-layout-components-bootstrap-alternatives.md

---

> title: Layout Components in Telerik UI for Blazor

Reviewer: I guess your purpose is to guide the developer in choosing the most suitable component, so you can state this directly.
```suggestion
title: Choosing the Right Layout Components for Your App
```

Author: Actually we refrain from consulting customers what to use. I also aimed for a shorter title that starts with a meaningful and relevant word.

slug: common-kb-layout-component-comparison
position: 
tags: telerik, blazor, layout, bootstrap
ticketid:
res_type: kb
---

## Environment

<table>
    <tbody>
        <tr>
            <td>Product</td>
            <td>
                UI for Blazor, <br />
                Form for Blazor, <br />
                GridLayout for Blazor, <br />
                MediaQuery for Blazor, <br />
                Splitter for Blazor, <br />
                StackLayout for Blazor, <br />
                TileLayout for Blazor
            </td>
        </tr>
    </tbody>
</table>

## Description

This KB article answers the following questions:

* What are the differences between the Dialog and the Window component? Dialog vs Window comparison.
* What are the differences between the Popover and the Tooltip? Popover vs Tooltip comparison.
* When should you use one popup component over another?

## Solution

### Bootstrap Framework Alternatives

If you do not want to use the Bootstrap framework to create your layouts, there are several components from the Telerik UI for Blazor suite that can serve similar purposes. Below you will find some examples to get you started:

> * The [Form](https://demos.telerik.com/blazor-ui/form/overview) supports built-in horizontal and vertical orientation of the labels and editors. You can also split form items in groups or organize the form layout in columns. Finally, the Form can generate input components for your model automatically.

Reviewer: Consider using a table that combines both lists. It should improve the readability.

Author: I agree that a table will improve readability, but this content does not present tabular data like the one here:

https://docs.telerik.com/blazor-ui/knowledge-base/common-popup-component-comparison#solution

If we think that lists are not readable enough, we should come up with a design change for them.
